Psychtoolbox/PsychGLImageProcessing/CreateProceduralColorGrating.m

```diff
@@ -0,0 +1,87 @@
+function [id, rect, shader] = CreateProceduralColorGrating(windowPtr, width, height, ...
+    color1, color2, radius, normalise)
+% [id, rect, shader] = CreateProceduralColorGrating(windowPtr, width, height [, color1=[1 0 0]]  [, color2=[0 1 0]] [, radius=0])
```

Comment:
OK, I used the wrap comment function in the MATLAB editor

Reply:
Sometimes i find the wrapping to be more harmful than helpful, e.g., for the function syntax line, as opposed to the help text body.

What i meant is, e.g., have a newline between some "section headers" and parameters, and inbetween parameters. The doc generator has the difficult job of transforming a total mess of inconsistent help text styles by different people in over 20 years into something half-way coherent. And it is written in Python, using all kinds of text parsing stuff for which my very basic Python knowledge and time is insufficient to improve upon it. I guess the best guideline is to look at function help texts, where the online docs look not too awful, and then emulate that style. Not sure how well or not the doc generator will parse this help text. May be fine or may be not fine.

Reply:
The other thought is that we should probably strive to provide some mathematical description in the help text of CreateProceduralXXXX() functions of what the shader does and how the different parameters factor in. I did this for some of the functions, for one i just made a reference to the Wikipedia definition that is implemented.

In the end, only the GLSL code is the ground truth, but if we want people to use these functions without having to learn all the GLSL themselves, it would be good to have some reasonably accurate mathematical definition in the help text. Cfe.:

https://github.com/kleinerm/Psychtoolbox-3/blob/033b6a0409578c2aa898520b66cf71e07faf8a33/Psychtoolbox/PsychGLImageProcessing/CreateProceduralSineGrating.m#L51

and multiple such definitions starting at:

https://github.com/kleinerm/Psychtoolbox-3/blob/033b6a0409578c2aa898520b66cf71e07faf8a33/Psychtoolbox/PsychGLImageProcessing/CreateProceduralSmoothedApertureSineGrating.m#L60

Reply:
OK, I have given more clear text headers for each section in the help text. It seems text on its own line gets converted into markdown headings, so I used that. It seems that they are markdown files so I used markdown lists to delineate input/output options; I haven't seen this style in the docs but in theory it should look much better for the online docs (I use markdown for everything and I think it looks clearer even in plain text, but that is 'IMO'). Let me know if you think that is not acceptable and I'll remove the list styling.

I also added algorithmic details on the shader, the basic functions are actually GLSL functions smoothstep and mix so I gave a description of them and links to the online documentation.

Reply:
Hm, I just tested the `PTB-wikify` script and it seems to escape markdown lists (line 180), but the plain text still looks good to me, you get an unordered list:

\* id = texture pointer.
\* rect = default position rect.

But this is inconsistent to previous help docs...

Reply:
Whatever makes the PTB-wikify script happy and doesn't look to hideous if one just types "help foo" in Matlab/Octave. So if this makes it look better, good.

But generally, consistency is good.